cli/grammar.go

```diff
@@ -103,6 +104,44 @@ func (k KeysGrammar) Run(ctx *kong.Context, tx *buntdb.Tx) error {
 	})
 }
 
+type SearchGrammar struct {
+	Pattern string `arg:"" help:"the match pattern "`
+	Delete  bool   `optional:"" short:"c" help:"delete all keys found (DANGEROUS)"`
+}
+
+func (s SearchGrammar) Run(ctx *kong.Context, tx *buntdb.Tx) error {
+	var keys []string
+	err := tx.AscendKeys("*", func(key, value string) bool {
+		if strings.Contains(value, s.Pattern) {
+			keys = append(keys, key)
+			fmt.Fprintf(ctx.Stdout, "%v : %v\n", key, value)
+		}
+		return true
+	})
+
+	if err != nil {
+		return err
+	}
+
+	if !s.Delete {
+		return nil
+	}
+
+	fmt.Fprintf(ctx.Stdout, "\n\n-------------------\n\nFound %d keys, written above. \nSleeping 10 seconds before we delete.", len(keys))
+	for n := 0; n != 10; n++ {
+		time.Sleep(1 * time.Second)
+		fmt.Fprintf(ctx.Stdout, ".")
+	}
+
+	for _, k := range keys {
+		_, err := tx.Delete(k)
+		if err == nil {
+			fmt.Fprintln(ctx.Stdout, "Deleted: %v", k)
```

Comment:
formatting directive %v?

Reply:
> Hi @yunginnanet thanks for your work. Do we need `search` command while there has been a `keys` command already? Or we can just add the `delete` option to `keys` command?

Keys only checks for a pattern against *keys*, not the contents of values

Reply:
I see